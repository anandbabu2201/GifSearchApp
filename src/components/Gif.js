import React, { Component } from "react";
import PropTypes from "prop-types";

class Gif extends Component {

  constructor(props){
    super(props);
    this.imgRef= React.createRef();
    this.state={
      playGif:false
    }
  }

  getExtenstion(src){
    let ext ;
    ext = src.toLowerCase().substring(src.lastIndexOf(".") + 1).trim();
    return ext;
  }

  loadGif =()=>{
    let {images} = this.props.gif
    if(this.imgRef && this.imgRef.current && this.getExtenstion(this.imgRef.current.src) === "jpg") {
     this.imgRef.current.src=images.downsized.url
     this.setState({playGif:true})
    }
     else {
     this.imgRef.current.src=images["480w_still"].url;
     this.setState({playGif:false})
     }
  }

  render() {
    let { title, images} = this.props.gif
    return (<div onClick={this.loadGif} className="eachBlock widescreen-4 desktop-4 tablet-landscape-3 tablet-2 mobile-landscape-2 mobile-1">
      <img  ref={this.imgRef} className="eachGif pause" alt={title}
      src={images["480w_still"].url}/>
     {!this.state.playGif?<div className="btn"></div>:''}
    </div>)

  }
}

Gif.propTypes = {
  gif: PropTypes.object.isRequired
};

export default Gif;
