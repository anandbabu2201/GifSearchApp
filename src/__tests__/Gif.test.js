import React from 'react';
import { mount } from 'enzyme';
import Gif from '../components/Gif';
let wrapper;

describe('Testing Gif component',() => {
  let props={gif:{
    id:1,
    title:"test",
    images:
      {
        "480w_still":{url: "sampleimage.jpg"},
        "downsized":{url: "simpletext.gif" }
       },
    }};
  wrapper = mount(<Gif {...props}/>);
  it('renders atleast one gif', () => {
    expect(wrapper.find('div.eachBlock')).toHaveLength(1);
  });
  it('sould play gif on clicking on play button', () => {
    wrapper.find('div.eachBlock').simulate('click');
    expect(wrapper.find('img').prop('src')).toBe('sampleimage.jpg')
  });
  it('sould stop play gif on clicking again play button', () => {
    wrapper.find('div.eachBlock').simulate('click');
    wrapper.find('div.eachBlock').simulate('click');
  });
});

