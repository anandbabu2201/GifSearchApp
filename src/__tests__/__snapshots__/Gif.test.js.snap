// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Gif component renders GifList component 1`] = `ShallowWrapper {}`;
